common/sgx/quote.c

```diff
@@ -481,13 +481,21 @@ oe_result_t oe_verify_quote_internal_with_collaterals(
         // time if one was not provided.
         if (validation_time == NULL)
         {
+            oe_sgx_endorsements_t* sgx_endorsements =
+                (oe_sgx_endorsements_t*)endorsements;
             OE_CHECK_MSG(
                 oe_datetime_from_string(
-                    collaterals_body->creation_datetime,
-                    sizeof(collaterals_body->creation_datetime),
+                    (const char*)sgx_endorsements
+                        ->items[OE_SGX_ENDORSEMENT_FIELD_CREATION_DATETIME]
+                        .data,
+                    sgx_endorsements
+                        ->items[OE_SGX_ENDORSEMENT_FIELD_CREATION_DATETIME]
+                        .size,
```

Comment:
If I understand right, since the verify function can be called on a machine that is not an SGX machine, this code is not safe... if it's run on a big endian machine, the .size value will be backwards since you didn't go through htonl (when sending) and ntohl (on receipt) or similar transform.   In short, when writing cross-platform code that parses messages sent across a network, it's never safe to simply cast a buffer to a struct that contains integers larger than 8 bits.

Ok to address in a separate PR if you file an issue to track this.

Reply:
See #2266